openedx/core/djangoapps/cookie_metadata/tests/test_cookie_name_change.py

```diff
@@ -0,0 +1,110 @@
+"""
+Test Modle to test CookieNameChange class
+"""
+from unittest.mock import Mock
+
+from django.test import TestCase
+
+
+from ..middleware import CookieNameChange
+
+
+class TestCookieNameChange(TestCase):
+    """
+    Test class for CookieNameChange
+    """
+
+    def setUp(self):
+        super().setUp()
+        self.mock_response = Mock()
+        self.cookie_name_change_middleware = CookieNameChange(self.mock_response)
+        self.mock_request = Mock()
+
+        self.old_value = "." * 100
+        self.old_key = 'a'
+        self.extra_cookies = {
+            "_b": "." * 13,
+            "_c_": "." * 13,
+            "a.b": "." * 10,
+        }
+        self.old_dict = {
+            self.old_key: self.old_value,
+        }
+
+        self.expand_settings = {
+            "old_name": self.old_key,
+            "new_name": "b",
+            "old_domain": "old_domain.com",
+        }
+
+    def test_cookie_swap(self):
+        """Check to make sure self.Middleware correctly swaps keys"""
+
+        self.old_dict.update(self.extra_cookies)
+
+        self.mock_request.COOKIES = self.old_dict.copy()
+
+        with self.settings(
+            COOKIE_NAME_CHANGE_ACTIVATE_EXPAND_PHASE=True
+        ), self.settings(COOKIE_NAME_CHANGE_EXPAND_INFO=self.expand_settings):
+            self.cookie_name_change_middleware(self.mock_request)
```

Comment:
Neat, I didn't know you could list multiple context managers in a `with` statement.